components/chainSelect/ChainSelect.tsx

```diff
@@ -133,25 +135,26 @@ const ChainSelect = () => {
 
     try {
       const chainData = await getChainFromRegistry(chainOption.path);
-      const assets = await getAssetsFromRegistry(chainOption.path);
-      const firstAsset = assets[0];
+      const registryAssets = await getAssetsFromRegistry(chainOption.path);
+      const firstAsset = registryAssets[0];
```

Comment:
Let's assert `registryAssets.length >= 1` with a helpful message here before using it. We never know what we will get from the remote source.